File: Demo/Sources/ViewControllers/Choise/ViewModels/ChoiseViewModel.m
//
//  Globus
//

#import "ChoiseViewModel.h"

@implementation ChoiseViewModel

+ (instancetype)viewModelWithTitle:(NSString*)title viewController:(Class)viewController {
    ChoiseViewModel* viewModel = [ChoiseViewModel new];
    viewModel.title = title;
    viewModel.viewController = viewController;
    return viewModel;
}

- (GLBViewController*)instantiateViewController {
    return [_viewController instantiate];
}

@end

File: Demo/Sources/ViewControllers/Choise/ViewModels/ChoiseViewModel.swift
//
//  Globus
//

import Globus

class ChoiseViewModel {
    
    var title: String = ""
    
    public init(title: String) {
        self.title = title
    }
    
    public func instantiateViewController() -> GLBViewController? {
        return nil
    }
    
}

class ChoiseControllerViewModel < ViewControllerClass : GLBViewController > : ChoiseViewModel {
    
    public override func instantiateViewController() -> GLBViewController? {
        return ViewControllerClass.instantiate()
    }
    
}
